lib/truffle/truffle/cext.rb

```diff
@@ -1152,18 +1154,12 @@ def rb_funcallv(recv, meth, args)
   end
 
   def rb_funcall(recv, meth, n, *args)
-    old_c_block = Thread.current[:__C_BLOCK__]
-    begin
-      block = Thread.current[:__C_BLOCK__]
-      Thread.current[:__C_BLOCK__] = nil
-      if block
-        recv.__send__(meth, *args, &block)
-      else
-        recv.__send__(meth, *args)
-      end
-    ensure
-      Thread.current[:__C_BLOCK__] = old_c_block
-    end
+    # see #call_with_thread_locally_stored_block
+    thread_local_block           = Thread.current[:__C_BLOCK__]
```

Comment:
The `begin` here allowed that if `Thread.current[:__C_BLOCK__]` failed for some reason (it shouldn't) then the fiber-local would not be set to nil (if `Thread.current[:__C_BLOCK__] = thread_local_block` works).
But this simplifies things and it seems things are badly broken if reading a fiber-local raises.